src/lib/mina/transaction.ts

```diff
@@ -276,6 +280,40 @@ type RejectedTransaction = Pick<
   errors: string[];
 };
 
+interface TransactionPromise extends Promise<Transaction> {
+  sign(...args: Parameters<Transaction['sign']>): TransactionPromise;
+  send(): PendingTransactionPromise;
+}
+
+function toTransactionPromise(
+  getPromise: () => Promise<Transaction>
+): TransactionPromise {
+  const pending = getPromise().then();
```

Comment:
why are you doing `.then()` here?

Reply:
Incase the `getPromise` returns a promise that was defined out of the scope of the `getPromise` fn, this ensures that we don't mutate / attach the methods to that original promise. Given how this code uses `toTransactionPromise`/`toPendingTransactionPromise`, forgoing the `then` wouldn't problematic. Still may be a worthwhile safeguard. Should I remove?

Reply:
aah I get it now. don't mutate your input values, excellent practice 👍🏻 since the overhead is negligible, leave it in!